tests/sentry/api/endpoints/test_user_notification_providers.py

```diff
@@ -0,0 +1,137 @@
+            value=NotificationSettingsOptionEnum.ALWAYS.value,
+        )
+        NotificationSettingProvider.objects.create(
+            user_id=self.user.id,
+            scope_type=NotificationScopeEnum.ORGANIZATION.value,
+            scope_identifier=self.organization.id,
+            type=NotificationSettingEnum.WORKFLOW.value,
+            provider=ExternalProviderEnum.EMAIL.value,
+            value=NotificationSettingsOptionEnum.ALWAYS.value,
+        )
+        NotificationSettingProvider.objects.create(
+            user_id=other_user.id,
+            scope_type=NotificationScopeEnum.ORGANIZATION.value,
+            scope_identifier=self.organization.id,
+            type=NotificationSettingEnum.ISSUE_ALERTS.value,
+            provider=ExternalProviderEnum.SLACK.value,
+            value=NotificationSettingsOptionEnum.ALWAYS.value,
+        )
+
+        response = self.get_success_response("me", type="alerts").data
+        assert len(response) == 2
+        slack_item = next(item for item in response if item["provider"] == "slack")
+        email_item = next(item for item in response if item["provider"] == "email")
+
+        assert slack_item["scopeType"] == "organization"
+        assert slack_item["scopeIdentifier"] == str(self.organization.id)
+        assert slack_item["user_id"] == str(self.user.id)
+        assert slack_item["team_id"] is None
+        assert slack_item["value"] == "always"
+        assert slack_item["type"] == "alerts"
+        assert slack_item["provider"] == "slack"
+
+        assert email_item["provider"] == "email"
+
+        response = self.get_success_response("me").data
+        assert len(response) == 3
+
+    def test_invalid_type(self):
+        response = self.get_error_response(
+            "me",
+            type="invalid",
+            status_code=status.HTTP_400_BAD_REQUEST,
+        )
+        assert response.data["type"] == ["Invalid type"]
+
+
+@control_silo_test(stable=True)
+class UserNotificationProvidersPutTest(UserNotificationProvidersBaseTest):
+    method = "PUT"
+
+    def setUp(self):
+        super().setUp()
+        self.login_as(self.user)
+
+    def test_simple(self):
+        response = self.get_success_response(
+            "me",
+            user_id=self.user.id,
+            scope_type="organization",
+            scope_identifier=self.organization.id,
+            type="alerts",
+            status_code=status.HTTP_201_CREATED,
+            value="always",
+            provider=["slack"],
+        )
+        assert NotificationSettingProvider.objects.filter(
+            user_id=self.user.id,
+            scope_type=NotificationScopeEnum.ORGANIZATION.value,
+            scope_identifier=self.organization.id,
+            type=NotificationSettingEnum.ISSUE_ALERTS.value,
+            value=NotificationSettingsOptionEnum.ALWAYS.value,
+            provider=ExternalProviderEnum.SLACK.value,
+        ).exists()
+        assert len(response.data) == 3
```

Comment:
can you assert on behavior that modifies / adds the other two providers modified by the request?